frontend/app/work_packages/services/work-package-field-service.js

```diff
@@ -323,6 +335,9 @@ module.exports = function(
   function format(workPackage, field) {
     var schema = getSchema(workPackage);
     if (field === 'date') {
+      if(workPackage.embedded.type.props.name === 'Milestone') {
+        return workPackage.props.startDate;
```

Comment:
@myabc we refactored those calls to ```isMilestone()``` (currently it check the href to eq /api/v3/types/4, but it is it one place only until we figure out how to check properly)